stock_restrict_lot_update/models/stock_move.py

```diff
@@ -0,0 +1,34 @@
+# Copyright 2023 Ooops404
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, fields, models
+from odoo.tools.safe_eval import safe_eval
+
+
+class StockMove(models.Model):
+    _inherit = "stock.move"
+
+    change_restrict_lot = fields.Boolean(
+        compute="_compute_change_restrict_lot",
+        help="If the product is in the domain",
+    )
+
+    @api.depends("product_id", "company_id")
+    def _compute_change_restrict_lot(self):
+        for move in self:
+            product_domain = safe_eval(
+                move.sudo().company_id.enforce_lot_restriction_product_domain
+            )
+            # if product is not in the enforcement domain, user can change the lot
+            move.change_restrict_lot = not move.product_id.filtered_domain(
+                product_domain
+            )
+
+    def write(self, vals):
+        res = super().write(vals)
+        if "restrict_lot_id" in vals:
```

Comment:
Could you add a comment about what this code does?

Reply:
Done!